app/controllers/shops_controller.rb

```diff
@@ -60,14 +60,13 @@ def change_status
     @shop = Shop.find(params[:shop_id])
     authorize! :show, @shop
     @appointment = Appointment.find(params[:appointment])
-    case params[:status]
-    when 'ready'
+    if @appointment.booked
       @appointment.ready_for_pickup!
       flash[:success] = 'The appointment has been updated to ready for pickup!'
       return redirect_to company_shop_index_appointment_path(@company, @shop)
-    when 'finished'
+    elsif @appointment.ready_for_pickup
```

Comment:
don't you mean @appointment.ready_for_pickup?

also this is not what I asked. You need to test the action:  @appointment.finished! because it can fail. and have a good / bad flash message depending on outcome